### PR TITLE
Allow `MediaSource` to be passed in to `URL.createObjectURL`

### PR DESCRIPTION
**Pull request by yilinwei**
See [here](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL_static#parameters).

### CODE REVIEW

File: dom/src/main/scala/org/scalajs/dom/URL.scala [hunk 1]
```diff
@@ -16,9 +17,9 @@ object URL extends js.Object {
 
   /** The URL.createObjectURL() static method creates a DOMString containing an URL representing the object given in
     * parameter. The URL lifetime is tied to the document in the window on which it was created. The new object URL
-    * represents the specified File object or Blob object.
+    * represents the specified File, Blob or MediaSource object.
     */
-  def createObjectURL(blob: Blob): String = js.native
+  def createObjectURL(obj: Blob | File | MediaSource): String = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/URL.scala:22]: `File` extends `Blob` so I don't think we need to add it as another case.

Comment (yilinwei) [reply dom/src/main/scala/org/scalajs/dom/URL.scala:22]: Sure I don't mind either way; what do you want to do with the docstring?

Comment (armanbilge) [reply dom/src/main/scala/org/scalajs/dom/URL.scala:22]: You can either copy it, or just leave it on the first overload. I think that's what we've done in other places.

File: dom/src/main/scala/org/scalajs/dom/URL.scala [hunk 2]
```diff
@@ -16,9 +16,12 @@ object URL extends js.Object {
 
   /** The URL.createObjectURL() static method creates a DOMString containing an URL representing the object given in
     * parameter. The URL lifetime is tied to the document in the window on which it was created. The new object URL
-    * represents the specified File object or Blob object.
+    * represents the specified File, Blob or MediaSource object.
     */
-  def createObjectURL(blob: Blob): String = js.native
+  def createObjectURL(obj: Blob): String = js.native
```
Comment (armanbilge) [inline dom/src/main/scala/org/scalajs/dom/URL.scala:21]: Can we restore this to `blob: Blob`? thanks!

**Pull request merged**